source/fundamentals/crud/read-operations/retrieve.txt

```diff
@@ -0,0 +1,242 @@
+.. _charp-retrieve:
+
+=============
+Retrieve Data
+=============
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to retrieve data from your MongoDB
+collections using the ``Find()`` method.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
+from the :atlas:`sample datasets </sample-data>` provided by Atlas. See the 
+:ref:`<csharp-quickstart>` to learn how to create a free MongoDB cluster and 
+load this sample data.
+
+.. literalinclude:: ../../../includes/code-examples/Restaurant.cs
+   :language: csharp
+   :copyable:
+   :dedent:
+
+.. _csharp-retrieve-find:
+
+Find Operation
+--------------
+
+You can use the ``Find()`` method to retrieve documents from a collection. 
+The ``Find()`` method takes a **query filter**, that specifies your
+search criteria, and returns all matching documents as an ``IFindFluent`` interface.
+
+Find One Document
+~~~~~~~~~~~~~~~~~
+
+To find a single document in a collection, pass a query filter that specifies the
+criteria of the document you want to find, then append the ``FirstOrDefault`` or 
+``FirstOrDefaultAsync`` method. If more than one document match the query
+filter these methods return the *first* matching document.
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: find-one-async
+
+      .. code-block:: csharp
+         :copyable: true
+
+         var restaurants = await _restaurantsCollection.Find(filter).FirstOrDefaultAsync();
+
+   .. tab:: Synchronous
+      :tabid: find-one-sync
+
+      .. code-block:: csharp
+         :copyable: true
+
+         var restaurants = _restaurantsCollection.Find(filter).FirstOrDefault();
+
+To see a full example of using ``Find()`` to find a single document, see
+:ref:`csharp-retrieve-additional-information`.
+
+Find Multiple Documents
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Find Documents Using A Cursor
```

Comment:
Issue: I'm not sure the reader can avoid using a cursor (I believe the "ToList" method [just iterates over all documents in a cursor and adds them to a list](https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/IAsyncCursor.cs#L380-L385), not comprehensive research but I believe this is how it works in other drivers we've looked at).

Consider rephrasing elements of this section to be more about iterating through retrieved documents rather than "Find documents using a cursor/list" (as everything is using a cursor).

Reply:
That's a good point with cursors that I wasn't aware of. 

I rephrased this section to focus more on iteration and the methods used, rather that making a distinction between cursor and list.